server/routes/statistics.js

```diff
@@ -236,17 +238,22 @@ module.exports = function (config) {
 			return resolveOperatorsDataPrint(unresolvedQuery, flowOperatorsCsv);
 		}),
 		'get-flow-rejections-data': function (unresolvedQuery) {
+			var data = { rows: [] }, page, queryData;
 			return rejectionsQueryHandler.resolve(unresolvedQuery)(function (query) {
-				var data = { rows: [], pageCount: 1 };
-				data.pageCount = getPage(data.rows, query.page);
-				/* TODO inflate data.rows with real results
-					expected structure:
-					[
-					['Reason 1, Reason 2', '*', 3, 'John Smith', 'Revision', '02/03/2017', 'Super inc.'],
-					['Reason', '', 0, 'John Doe', 'Precal', '01/04/2017', 'Bar']
-					]
-				 */
+				page = ensureNumber(query.page);
+				queryData = data;
+				return getRejectionReasons.count(queryData);
+			}).then(function (count) {
+				var offset, limit;
+				offset = (page - 1) * itemsPerPage;
+				if (page * itemsPerPage < count) limit = itemsPerPage;
+				data.pageCount = ensureNumber(Math.ceil(count / itemsPerPage));
+				return getRejectionReasons.find(queryData, offset, limit);
+			}).then(function (reasons) {
+				data.rows = parseRejectionsForView(reasons);
 				return data;
+			}, function (err) {
```

Comment:
Actually this error callback should not be needed. Like in case of other controller handlers we should return a promise or the final data... the controller generic mechanisms should catch errors.